organizations/tests.py

```diff
@@ -39,3 +49,57 @@ def test_superuser_can_list_organizations(self):
         self.user.save()
         response = self.client.get(reverse("organizations-list"))
         self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_updating_organization_recreates_user_instance_cache(self):
+        from users.models import User
+        from rest_framework.test import APIClient
+        from plio.tests import get_new_access_token
+
+        superadmin_client = APIClient()
+        superadmin_user = User.objects.create(mobile="+919988776655", is_superuser=True)
+        superadmin_access_token = get_new_access_token(
+            superadmin_user, self.application
+        )
+        superadmin_client.credentials(
+            HTTP_AUTHORIZATION="Bearer " + superadmin_access_token.token
+        )
+
+        # verify cache data doesn't exist by default
+        cache_key_name = get_cache_key(self.user)
+
+        # make a get request
+        self.client.get(reverse("users-detail", kwargs={"pk": self.user.id}))
+        # verify cache data exists as we made a GET request for user details
+        self.assertEqual(len(cache.keys(cache_key_name)), 1)
+        self.assertEqual(len(cache.get(cache_key_name)["organizations"]), 0)
+
+        # associate the current user with the organization
+        OrganizationUser.objects.create(
+            organization=self.organization_2, user=self.user, role=self.org_admin_role
+        )
+
+        # check cache after updating organization user association
+        # this time the user should have an organization associated
+        self.assertEqual(len(cache.get(cache_key_name)["organizations"]), 1)
+        self.assertEqual(
+            cache.get(cache_key_name)["organizations"][0]["name"],
+            self.organization_2.name,
+        )
```

Comment:
shouldn't this be a part of `OrganizationUserTestCase`?

Reply:
I think you can leave out the assert part here - and leave everything else. And use the version with assert under `OrganizationUserTestCase`. 

Also, I don't seem to be able to understand why this is working - as in, without a GET request first, why is the cache for the user getting updated? Upon updating organization user, the cache gets invalidated. But where is it getting filled up again with the updated value?

Reply:
as you mentioned in the docs as well, the user cache should be deleted but not updated right?
<img width="992" alt="Screenshot 2021-09-24 at 3 52 05 PM" src="https://user-images.githubusercontent.com/13185699/134659533-3242e796-2b79-4665-8eed-6e8e6c8fa60f.png">
he

Reply:
@dalmia I think your full comment didn't come through. can you share again?

Yes, in case related instances are updated, the primary instance cache gets deleted and not re-created. The cache for an instance is created in the corresponding serializer. Since updating related instance doesn't call the primary instance serializer, hence the update cache doesn't happen. Only deletion.

Reply:
hey sorry I saw your previous comment just now.

What you wrote makes sense and that's my understanding as well. I'm trying to figure out why it's working (updating automatically). That's weird. Will get back to you.